pkg/ddl/notifier/subscribe.go

```diff
@@ -133,7 +135,8 @@ func (n *DDLNotifier) RegisterHandler(id HandlerID, handler SchemaChangeHandler)
 	}
 
 	if _, ok := n.handlers[id]; ok {
-		panic(fmt.Sprintf("HandlerID %d already registered", id))
+		logutil.BgLogger().Error("HandlerID already registered", zap.Int("id", int(id)))
```

Comment:
This is possible in tests, so I changed it to an error log. Maybe we should only allow it in tests?

Reply:
By just code review I didn't find where the duplicated register happens. Can you provide more details?

If there are duplicated register, maybe it's in fact a data race, and golang will panic on map type race modification

Reply:
```go
func TestVersion(t *testing.T) {
	store, dom := testkit.CreateMockStoreAndDomain(t)
	testKit2 := testkit.NewTestKit(t, store)
	testKit := testkit.NewTestKit(t, store)
	testKit.MustExec("use test")
	testKit.MustExec("create table t1 (c1 int, c2 int)")
	testKit.MustExec("analyze table t1 all columns")
	do := dom
	is := do.InfoSchema()
	tbl1, err := is.TableByName(context.Background(), model.NewCIStr("test"), model.NewCIStr("t1"))
	require.NoError(t, err)
	tableInfo1 := tbl1.Meta()
	h, err := handle.NewHandle(
		testKit.Session(),
		testKit2.Session(),
		time.Millisecond,
		is,
		do.SysSessionPool(),
		do.SysProcTracker(),
		do.DDLNotifier(),
		do.NextConnID,
		do.ReleaseConnID,
	)
```

Here we created two handlers.

Reply:
Updated.